### PR TITLE
[SRVCOM-1329] Add metrics for openshift knative operator and ingress

### PR DESCRIPTION
**Pull request by skonto**
- Enables metrics for [openshift knative operator ](https://gist.github.com/skonto/b1012b47ef9818d12a7d34ffed9fb40f)and [ingress](https://gist.github.com/skonto/b6c1ef4c5d4e5ee74712ea039d06ab28). 
- Resources will be managed by olm as in the case of the serverless operator.
- Adds e2e tests for these metric endpoints.
- Endpoints are not secured as they do not disclose critical info.

### CODE REVIEW

File: test/monitoringe2e/helpers.go [hunk 1]
```diff
@@ -41,6 +41,15 @@ var (
 		"mt_broker_ingress_go_mallocs",
 		"sugar_controller_go_mallocs",
 	}
+
+	serverlessComponentQueries = []string{
+		// Checks if openshift-knative-operator metrics are served
+		"knative_operator_go_mallocs",
+		// Checks if knative-opeshift metrics are served
+		"controller_runtime_active_workers{controller=\"knativeserving-controller\"}",
```
Comment (skonto) [inline test/monitoringe2e/helpers.go:49]: We implicitly check this via the knative_up metric elsewhere but does not hurt to verify here.

File: templates/csv.yaml [hunk 2]
```diff
@@ -400,7 +400,12 @@ spec:
                   # This reference will be replaced in local builds and CI via hack/lib/catalogsource.bash.
                   image: registry.ci.openshift.org/openshift/openshift-serverless-nightly:knative-openshift-ingress
                   imagePullPolicy: Always
+                  ports:
+                    - containerPort: 9090
+                      name: metrics
                   env:
+                    - name: METRICS_DOMAIN
+                      value: knative.dev/serving-ingress
```
Comment (markusthoemmes) [inline templates/csv.yaml:408]: Why not `knative.dev/serving`?

Comment (skonto) [reply templates/csv.yaml:408]: no preference just because it is for ingress.

**Comment by skonto** [comment]
@markusthoemmes ready to merge if there are no more comments.

**Comment by markusthoemmes** [comment]
I'd like to see this reviewed by somebody else to spread knowledge about our monitoring pieces further. It's currently very constrained to just @skonto and myself.

**Comment by nak3** [comment]
LGTM

File: templates/csv.yaml [hunk 3]
```diff
@@ -398,7 +398,12 @@ spec:
                   # This reference will be replaced in local builds and CI via hack/lib/catalogsource.bash.
                   image: registry.ci.openshift.org/openshift/openshift-serverless-nightly:knative-openshift-ingress
                   imagePullPolicy: Always
+                  ports:
+                    - containerPort: 9090
+                      name: metrics
                   env:
+                    - name: METRICS_DOMAIN
+                      value: knative.dev/serving
```
Comment (matzew) [inline templates/csv.yaml:406]: do we need something like this as well for the `knative-operator`?

Comment (matzew) [reply templates/csv.yaml:406]: Ah, I think it's here: https://github.com/openshift-knative/serverless-operator/blob/a3b7c60f8542f8c232093d4abf0a2d7b81cd087b/olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml#L380-L386

File: test/monitoringe2e/helpers.go [hunk 4]
```diff
@@ -41,6 +41,15 @@ var (
 		"mt_broker_ingress_go_mallocs",
 		"sugar_controller_go_mallocs",
 	}
+
+	serverlessComponentQueries = []string{
+		// Checks if openshift-knative-operator metrics are served
+		"knative_operator_go_mallocs",
+		// Checks if knative-opeshift metrics are served
```
Comment (matzew) [inline test/monitoringe2e/helpers.go:48]: `knative-opeshift`  typo?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**